### ISSUE TITLE
sql: move `ALTER TYPE ... DROP VALUE ...` out of alpha

### ISSUE THREAD

**Issue by arulajmani** [cockroachdb/cockroach#61594]
**Is your feature request related to a problem? Please describe.**

Currently, `ALTER TYPE ... DROP VALUE` is supported as an alpha feature and to that end is gated behind the `enable_drop_enum_value` SESSION setting (or the corresponding `sql.defaults.drop_enum_value.enabled` CLUSTER setting. The main reason behind this gating this functionality behind a feature flag are the limitations described in https://github.com/cockroachdb/cockroach/issues/59807. Once those limitations are resolved, we should move this feature out of alpha. 

Followup as a result of #61519

Epic CRDB-2451

**Comment by rafiss** [comment]
@vy-ton is there a timeline for when alpha needs to be removed from this feature?

**Comment by vy-ton** [comment]
@rafiss this was de-scoped for 21.2 due to other priorities, but we can opportunistically tackle it after #59807 closes.

**Comment by rafiss** [comment]
Update: we will need to resolve https://github.com/cockroachdb/cockroach/issues/68361 first
